src/hotspot/share/memory/arena.hpp

```diff
@@ -134,6 +134,15 @@ class Arena : public CHeapObj<mtNone> {
   void* Amalloc(size_t x, AllocFailType alloc_failmode = AllocFailStrategy::EXIT_OOM) {
     x = ARENA_ALIGN(x);  // note for 32 bits this should align _hwm as well.
```

Comment:
The comment here is stale.

Reply:
> The comment here is stale.

Yes, I know. See my initial PR text.

I left it in since my first fix was rejected for being too complex, so I wanted to limit this patch to the absolute minimum needed. 

I do not think aligning the allocation size - or insisting on an aligned allocation size as the sister method `AmallocWords` does - is useful at all. Removing the comment just leaves the code, which leaves the reader to ask himself why it's in there. Leaving the comment at least states the reason the original author saw. We should either remove the code and the comment or leave both in place.

Reply:
The length alignment (either by adjustment or by precondition) is what lets the ARENA_ALIGN of _hwm be 32bit only in Amalloc, and not needed in AmallocWords.  Maybe you are suggesting dropping the size alignment and instead unconditionally aligning _hwm as needed for the operation?  Maybe that's the next RFE?

Reply:
Oh please no more RFEs unless Kim can review them.  This turned out much more difficult than we thought it would be!
This comment is true though, right?  For 32 bits if you do an AmallocWords then Amalloc, the _hwm will then be aligned to ARENA_ALIGN, ie 64 bits.  That's ok.  That seems like just what we want.  Although it seems like this latter comment says what I think this does better.  I'm fine with leaving the comment though.  Makes you think about it anyway.